src/runtime_src/core/pcie/driver/linux/xocl/subdev/qdma.c

```diff
@@ -438,24 +438,26 @@ static void free_channels(struct platform_device *pdev)
 	struct mm_channel *chan;
 	u32	write, qidx;
 	int i, ret = 0;
+	char	ebuf[MM_EBUF_LEN + 1];
```

Comment:
Stack size for kernel thread is generally very limited. On Linux, it's probably just 4K or 8K. It is a bad idea to put a big array on stack like this since you'll run out of stack pretty quick. So, please kzalloc() the buffer and you don't even need memset() later. Or use a much smaller buffer here since no one seems to consume the content after the call anyway?

Reply:
Hi @maxzhen MM_EBUF_LEN is defined as 256 bytes only and the same buffer is being used for all qdma channels in a loop. It is not tied with any single channel. So, buffer contents cleared using memset(). Even, kzalloc() used, buffer contents has to be cleared as it is. Please let me know if you still want to use kzalloc (dynamic) way. Thank you.

Reply:
I see why you need to memset to clear it. It is normally more efficient for the API to clear it internally since they know when it is needed and how much space it needs to be cleared. But it is what it is.
The concern is that 256 bytes sounds too big for kernel stack. Please dynamically alloc it or put it in some global data structure so that it does not show up in the stack.

Reply:
Hi @maxzhen I have updated the code to use dynamic alloc for this error buffer. This buffer is used only to store return message from qdma driver, and this buffer should be of some size, but not null.
For each channel, we are using this same buffer hence we are clearing the contents to make sure the buffer doesn't contain previous channel's return message.